src/utils/s3_mods.py

```diff
@@ -0,0 +1,148 @@
+    client = boto3.client("s3")
+    raz_client.configure_ranger_raz(client, ssl_file=config["s3"]["ssl_file"])
+    return client
+
+
+# Read a CSV file into a Pandas dataframe
+def rd_read_csv(filepath: str, **kwargs) -> pd.DataFrame:
+    """Reads a csv from s3 bucket into a Pandas Dataframe using pydoop. 
+    If "thousands" argument is not specified, sets it to ",". 
+    Allows to use any additional keyword arguments of Pandas read_csv method.
+
+    Args:
+        filepath (str): Filepath (Specified in config)
+        kwargs: Optional dictionary of Pandas read_csv arguments
+    Returns:
+        pd.DataFrame: Dataframe created from csv
+    """
+    # Open the boto3 client
+    s3_client = config["client"]
+    with s3_client.get_object(
+        Bucket=config["s3"]["s3_bucket"],
+        Key=filepath
+    )['Body'] as file:
+        
+        # If "thousands" argument is not specified, set it to ","
+        if "thousands" not in kwargs:
+            kwargs["thousands"] = ","
+
+        # Read the scv file using the path and keyword arguments
+        try:
+            df = pd.read_csv(file, **kwargs)
+        except Exception:
+            if "usecols" in kwargs:
+                rd_logger.info("Columns not found: " + str(kwargs["usecols"]))
+            rd_logger.error(f"Could not read specified file: {filepath}")
+
+            raise ValueError
+    return df
+  
+
+def rd_write_csv(filepath: str, data: pd.DataFrame) -> None:
+    """Writes a Pandas Dataframe to csv in s3 bucket
+
+    Args:
+        filepath (str): Filepath (Specified in config)
+        data (pd.DataFrame): Data to be stored
+    """
+    # Create an Unput-Output buffer
+    csv_buffer = StringIO()
+    
+    # Write the dataframe to the buffer in the CSV format 
+    df.to_csv(csv_buffer, header=True, date_format="%Y-%m-%d %H:%M:%S.%f+00", index=False)
+    
+    # "Rewind" the stream to the start of the buffer
+    csv_buffer.seek(0) 
+    
+    # Use the boto3 client from the config
+    s3_client = config["client"]
+    
+    # Write the buffer into the s3 bucket
+    _ = s3_client.put_object(
+        Bucket=config["s3"]["s3_bucket"], 
+        Body=csv_buffer.getvalue(), 
+        Key=filepath
+    )
+    return None
+
+
+def rd_load_json(filepath: str) -> dict:
+    """Function to load JSON data from s3 bucket
+    Args:
+        filepath (string): The filepath in Hue
+    """
+    # Use the boto3 client from the config
+    s3_client = config["client"]
+    
+    # Load the json file using the client method
+    with s3_client.get_object(Bucket=config["s3"]["s3_bucket"], Key=filepath)['Body'] as json_file:
+        datadict = json.load(json_file)
+    
+    return datadict
+
+
+def rd_file_exists(filepath: str, raise_error=False) -> bool:
+    """Function to check file exists in hdfs.
+
+        Args:
+            filepath (string) -- The filepath in s3
+
+        Returns:
+            Bool - A boolean value which is true if file exists
+    """
```

Comment:
Add type hint to raise_error param.
Update docstring to both fit the correct format, and include all parameters.
Also raises should be added to the docstring (this is true for  all docstrings)

Reply:
Added a type hint raise_error, updated the doc string. I did not add Raises, as we are not doing it anywhere.

Reply:
This function raises FileExistsError. I have updated the docstring.